overlord/snapshotstate/backend/backend.go

```diff
@@ -476,6 +463,49 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 	return nil
 }
 
+// shouldIncludeInSnapshot determines what dirs under 'snapDir' should be
+// included in the snapshot, if any.
+func shouldIncludeInSnapshot(snapDir string, rev snap.Revision, snapshot *client.Snapshot) (includeRevDir, includeCommonDir bool, err error) {
```

Comment:
I would rename this to `pathsForSnapshot()` and have it return a list of paths; I think this would allow you to simplify the code quite a bit.

Reply:
I think that's a good suggestion and will probably fit nicely with your PR. The new version of this function could check what paths should be included, compute the exclusions (your PR) and then return some unified result.